File: Movie/API/PersonService.swift
//
//  PersonService.swift
//  Movie
//
//

import Foundation
import Alamofire

class PersonService {
    
    func getDetails(id: Int) async throws -> Person {
        return try await AF.request(
            "\(ApiConstants.baseUrl)/person/\(id)?api_key=\(ApiConstants.apiKey)"
        )
        .serializingDecodable(Person.self, decoder: DateDecoder())
        .value
    }
    
    func getCredits(id: Int) async throws -> PersonCreditsResponse {
        return try await AF.request(
            "\(ApiConstants.baseUrl)/person/\(id)/movie_credits?api_key=\(ApiConstants.apiKey)"
        )
        .serializingDecodable(PersonCreditsResponse.self, decoder: DateDecoder())
        .value
    }
}

File: shared/src/commonMain/kotlin/com/mehmetaliocak/movie_app/api/PersonService.kt
package com.mehmetaliocak.movie_app.api

import com.mehmetaliocak.movie_app.models.Person
import com.mehmetaliocak.movie_app.models.PersonCreditsResponse
import com.rickclephas.kmp.nativecoroutines.NativeCoroutines
import io.ktor.client.HttpClient
import io.ktor.client.call.body
import io.ktor.client.request.get
import org.koin.core.component.KoinComponent
import org.koin.core.component.inject

class PersonService : KoinComponent {
    private val client: HttpClient by inject()

    @NativeCoroutines
    suspend fun getDetails(id: Long): Person {
        return client.get("person/$id").body()

    }
    @NativeCoroutines
    suspend fun getCredits(id: Long): PersonCreditsResponse {
        return client.get("person/$id/movie_credits").body()
    }
}